swift/ql/test/query-tests/Security/CWE-094/UnsafeJsEval.swift

```diff
@@ -201,17 +201,17 @@ func testSync(_ sink: @escaping (String) -> ()) {
 	let url = URL(string: "http://example.com/")
 
 	sink(localString) // GOOD: the HTML data is local
-	sink(try! String(contentsOf: URL(string: "http://example.com/")!)) // BAD: HTML contains remote input, may access local secrets
-	sink(try! String(contentsOf: url!)) // BAD
+	sink(try! String(contentsOf: URL(string: "http://example.com/")!)) // $ Source=source1 $ MISSING: Alert HTML contains remote input, may access local secrets
+	sink(try! String(contentsOf: url!)) // $ Source=source2 $ MISSING: Alert
 
 	sink("console.log(" + localStringFragment + ")") // GOOD: the HTML data is local
-	sink("console.log(" + (try! String(contentsOf: url!)) + ")") // BAD
+	sink("console.log(" + (try! String(contentsOf: url!)) + ")") // $ Source=source3 $ MISSING: Alert
 
 	let localData = Data(localString.utf8)
-	let remoteData = Data((try! String(contentsOf: url!)).utf8)
+	let remoteData = Data((try! String(contentsOf: url!)).utf8) // $ Source=source4
 
 	sink(String(decoding: localData, as: UTF8.self)) // GOOD: the data is local
-	sink(String(decoding: remoteData, as: UTF8.self)) // BAD: the data is remote
+	sink(String(decoding: remoteData, as: UTF8.self)) // $ MISSING: Alert the data is remote
```

Comment:
I don't think there should be any `MISSING` tags in `testSync`.  The results are correctly flagged at all calls I think (e.g. lines 291, 305) for all sources (`source1` ... `source4`).  Even line 214 is found, but with the source on line 211 instead (which is fine).

Reply:
The query does not produce any alerts on line 214, and the existing comment suggested that it should have, so I assumed a `MISSING` annotation was in order. If no alert is expected at this line, we should simply remove the comment.

Reply:
Yep, it's a bit weird with this being a matrix test and the annotation system being informal until now.  The alerts are expected in practice at the callers (where they do all appear) and the `BAD` comments here are kindof in the wrong place, but perhaps expressed something valuable before we had source labels on the alerts at the callers.

Upshot is I believe you can remove all the "MISSING" comments from this function (`testSync`).

Reply:
I appreciate you tackling this query by the way, it really shows off the advantages of source labelling.

Reply:
Ok, I'll remove all `MISSING` comments inside `testSync`.

Reply:
:+1: